test/libyul/yulOptimizerTests/fullSuite/aztec.yul

```diff
@@ -281,80 +281,80 @@
 //         {
-//             mstore(validateJo__62, validateJo_c)
-//             let validateJo__120 := 0x220
-//             let validateJo_result_307 := and(validateJo_result, call(gas(), 7, validateJo_i_1306, validateJo__34, validateJo__62, validateJo__120, validateJo__52))
-//             let validateJo_result_308 := and(validateJo_result_307, call(gas(), 6, validateJo_i_1306, validateJo__120, validateJo__6, 0x260, validateJo__52))
-//             validateJo_result := and(validateJo_result_308, call(gas(), 6, validateJo_i_1306, validateJo__65, validateJo__6, 0x1e0, validateJo__52))
+//             mstore(validateJo_8, validateJo_c)
+//             let validateJo_11 := 0x220
+//             let validateJo_result_4 := and(validateJo_result, call(gas(), 7, validateJo_i_1, validateJo_3, validateJo_8, validateJo_11, validateJo_6))
+//             let validateJo_result_5 := and(validateJo_result_4, call(gas(), 6, validateJo_i_1, validateJo_11, validateJo, 0x260, validateJo_6))
+//             validateJo_result := and(validateJo_result_5, call(gas(), 6, validateJo_i_1, validateJo_9, validateJo, 0x1e0, validateJo_6))
 //         }
 //         if iszero(validateJo_result)
 //         {
-//             mstore(validateJo_i_1306, 400)
-//             revert(validateJo_i_1306, validateJo__34)
+//             mstore(validateJo_i_1, 400)
+//             revert(validateJo_i_1, validateJo_3)
 //         }
-//         validateJo_b := add(validateJo_b, validateJo__52)
+//         validateJo_b := add(validateJo_b, validateJo_6)
 //     }
 //     if lt(validateJo_m, validateJo_n)
 //     {
 //         validatePairing(0x64)
 //     }
-//     if iszero(eq(mod(keccak256(validateJo__24, add(validateJo_b, 0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffd60)), validateJo_gen_order), validateJo_challenge))
+//     if iszero(eq(mod(keccak256(validateJo_2, add(validateJo_b, 0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffd60)), validateJo_gen_order), validateJo_challenge))
 //     {
-//         mstore(validateJo_i_1306, 404)
-//         revert(validateJo_i_1306, 0x20)
+//         mstore(validateJo_i_1, 404)
+//         revert(validateJo_i_1, 0x20)
 //     }
-//     mstore(validateJo_i_1306, 0x01)
-//     return(validateJo_i_1306, 0x20)
-//     mstore(validateJo_i_1306, 404)
-//     revert(validateJo_i_1306, 0x20)
+//     mstore(validateJo_i_1, 0x01)
+//     return(validateJo_i_1, 0x20)
+//     mstore(validateJo_i_1, 404)
+//     revert(validateJo_i_1, 0x20)
 //     function validatePairing(t2)
 //     {
-//         let t2_x_1 := calldataload(t2)
+//         let t2_x := calldataload(t2)
 //         let _165 := 0x20
 //         let t2_x_2 := calldataload(add(t2, _165))
-//         let t2_y_1 := calldataload(add(t2, 0x40))
+//         let t2_y := calldataload(add(t2, 0x40))
 //         let t2_y_2 := calldataload(add(t2, 0x60))
 //         let _171 := 0x90689d0585ff075ec9e99ad690c3395bc4b313370b38ef355acdadcd122975b
```

Comment:
It looks like this should have a lower number, i.e. the empty prefix should be a valid prefix (but of course not a valid name).

Reply:
the reason why it didn't pick t2_y_1 for t2_y_2 in the line below is actually because all t2_y_1's are to be mapped to t2_y, so that means, the cleaner already knows that this name is "in use".

I would need to have two maps in order work around that. And I concluded (back in time) that it wasn't worth it (yet) :)

Reply:
Ok, I assumed you reported at line 353 only for more context. I intentionally did not rewrite identifiers that look only like a suffix (no characters before). I've changed that behavior now.